packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -313,39 +269,78 @@ export class TeamsBotV2Impl {
     throw new Error("Invalid programming language");
   }
 
-  private resolveScenarios(
-    actRoles: PluginActRoles[],
-    triggers: BotTrigger[],
-    hostType: HostType,
-    isM365: boolean | undefined
-  ): string[] {
-    const scenarios: string[] = [];
-    actRoles.map((actRole) => {
-      switch (actRole) {
-        case PluginActRoles.CommandAndResponse:
-          scenarios.push(TemplateProjectsScenarios.COMMAND_AND_RESPONSE_SCENARIO_NAME);
-          break;
-        case PluginActRoles.Notification:
-          if (hostType === HostTypes.AZURE_FUNCTIONS) {
-            scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
-            triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
-          }
-          if (hostType === HostTypes.APP_SERVICE) {
-            scenarios.push(TemplateProjectsScenarios.NOTIFICATION_RESTIFY_SCENARIO_NAME);
-          }
+  private resolveScenarios(ctx: Context, inputs: Inputs): string[] {
+    const templateScenarios: string[] = [];
+    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
+    const capabilities = solutionSettings.capabilities;
+    capabilities.map((capability: string) => {
```

Comment:
Is it possible that capabilities contains both bot and message extension?
In that case do we push default scenario twice?

Reply:
After merge templates there will be only one left. Also it's probable that msgext template won't share same template with bot in the future.